src/Parsers/IAST.cpp

```diff
@@ -71,6 +72,10 @@ void IAST::updateTreeHashImpl(SipHash & hash_state) const
 {
     auto id = getID();
     hash_state.update(id.data(), id.size());
+    if (auto * ast_with_alias = dynamic_cast<const ASTWithAlias *>(this))
```

Comment:
I don't think we need to hash aliases... because we use tree hash to glue identical expressions regardless to aliases.